### DISCUSSION TITLE
Is there any feature rich component built on top of react-markdown

### DISCUSSION THREAD

**Discussion by ZehuaZhang** [remarkjs/remark#1158] [Q&A] [answered]
Is there any feature rich component built on top of react-markdown, github, codesandbox, or npm?

Another question is that, I'm very new to this package, how would I go about to implement an image upload functionality, with drag and drop, or copy and paste. When dropping, it would says uploading, when done, it would return the cloud storage url.

And what about math, table, any good plugins for those?

**Comment by ChristianMurphy** [comment, marked as answer]
Heya @ZehuaZhang! 👋 

I'm not quite sure what you are looking for, your description covers three or four tangentially related things.
I'll answer them separately, feel free to add context on how they'd fit together in your use case.

> Is there any feature rich component built on top of react-markdown, github, codesandbox, or npm?

By component you mean an editor experience which can be embedded in an existing React/Vue/etc page?
If you are looking for rich text editing on top of remark/react-markdown/mdx:
* https://github.com/Milkdown/milkdown
* https://github.com/blocks/blocks
* https://github.com/bytedance/bytemd

Could be of interest.

> how would I go about to implement an image upload functionality, with drag and drop, or copy and paste.

It sounds like you want to build a content management system? (CMS)
Those are rather complex, and way outside the scope of what remark tries to solve, but there are some that build on remark.
* https://github.com/tinacms/tinacms
* https://github.com/decaporg/decap-cms

> When dropping, it would says uploading, when done, it would return the cloud storage url.

This doesn't sound like an editor or a CMS.
You may be looking for https://github.com/transloadit/uppy? /cc @Murderlon

> And what about math, table, any good plugins for those?

What about them?
What do you want to do with these syntax add ons?
